diff --git a/libexec/rc/rc.conf b/libexec/rc/rc.conf
index 0495e003b9c5..1c46b22f8389 100644
--- a/libexec/rc/rc.conf
+++ b/libexec/rc/rc.conf
@@ -194,6 +194,7 @@ firewall_nologports="135-139,445 1026,1027 1433,1434" # List of TCP/UDP ports
 firewall_nat_enable="NO"	# Enable kernel NAT (if firewall_enable == YES)
 firewall_nat_interface=""	# Public interface or IPaddress to use
 firewall_nat_flags=""		# Additional configuration parameters
+firewall_nat_rules=""		# File of parameters followed by firewall_nat_flags
 firewall_nat64_enable="NO"	# Enable kernel NAT64 module.
 firewall_nptv6_enable="NO"	# Enable kernel NPTv6 module.
 firewall_pmod_enable="NO"	# Enable kernel protocols modification module.
diff --git a/libexec/rc/rc.firewall b/libexec/rc/rc.firewall
index ee08a3252566..882bc74f84de 100644
--- a/libexec/rc/rc.firewall
+++ b/libexec/rc/rc.firewall
@@ -170,6 +170,9 @@ case ${firewall_type} in
 	case ${firewall_nat_enable} in
 	[Yy][Ee][Ss])
 		if [ -n "${firewall_nat_interface}" ]; then
+			if [ -n "${firewall_nat_rules}" -a -r ${firewall_nat_rules} ]; then
+				firewall_nat_flags="${firewall_nat_flags} $(cat ${firewall_nat_rules})"
+			fi
 			if echo "${firewall_nat_interface}" | \
 				grep -q -E '^[0-9]+(\.[0-9]+){0,3}$'; then
 				firewall_nat_flags="ip ${firewall_nat_interface} ${firewall_nat_flags}"
